packages/world/src/modules/init/implementations/getSignedMessageHash.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+
+bytes32 constant DELEGATION_TYPEHASH = keccak256(
+  "Delegation(address delegatee,bytes32 delegationControlId,bytes initCallData,uint256 nonce)"
+);
+
+function getSignedMessageHash(
+  address delegatee,
+  ResourceId delegationControlId,
+  bytes memory initCallData,
+  uint256 nonce,
+  address worldAddress
+) view returns (bytes32) {
+  bytes32 domainSeperator = keccak256(
+    abi.encode(keccak256("EIP712Domain(uint256 chainId,address verifyingContract)"), block.chainid, worldAddress)
```

Comment:
With this domain separator, signatures are unique across world addresses and chains. The EIP has optional `name` and `version` fields but i think they only makes sense if we were upgrading the signature scheme? https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator